### PR TITLE
Feature/dev testing

### PR DESCRIPTION
**Pull request by matthewpeterkort**

### CODE REVIEW

**Comment by bwalsh** [comment]
Getting the following error:

```
$ git pull origin feature/dev-testing
From github.com:bmeg/jsonschemagraph
 * branch            feature/dev-testing -> FETCH_HEAD
Already up to date.
RNB10644:jsonschemagraph walsbr$ go run cmds/gengraph/main.go
go: downloading google.golang.org/protobuf v1.30.0
go: downloading github.com/santhosh-tekuri/jsonschema/v5 v5.3.0
2023/07/13 10:42:32 Error reading file: /Users/walsbr/bmeg/jsonschemagraph/bmeg_schemas/same_a.yaml
[AlleleEffect Allele Pathway Protein Case Compound File GenePhenotypeAssociation Methylation Publication Command DrugResponse GeneSet MethylationProbe Program GeneOntologyTerm Interaction Phenotype Definitions Terms GenomicFeature Sample CopyNumberAlteration Gene GeneExpression Project ProteinStructure Exon Transcript]
Writing:  output/ensembl_exon.json.gz
ERROR ON FILE CREATE
ERROR ON FILE CREATE
ERROR ON FILE CREATE
LENGTH OF PROCCHAN 0graph schema validate error at
graph schema validate error at
graph schema validate error at
graph schema validate error at
graph schema validate error at
graph schema validate error at
Error getting file information: stat output/ensembl_exon.json.gz.OutEdge.json: no such file or directory
Error getting file information: stat output/ensembl_exon.json.gz.InEdge.json: no such file or directory
Error getting file information: stat output/ensembl_exon.json.gz.Vertex.json: no such file or directory
Writing:  output/ensembl_gene.json.gz
ERROR ON FILE CREATE
ERROR ON FILE CREATE
ERROR ON FILE CREATE
LENGTH OF PROCCHAN 0graph schema validate error at
graph schema validate error at
graph schema validate error at
graph schema validate error at
graph schema validate error at
graph schema validate error at
Error getting file information: stat output/ensembl_gene.json.gz.OutEdge.json: no such file or directory
Error getting file information: stat output/ensembl_gene.json.gz.InEdge.json: no such file or directory
Error getting file information: stat output/ensembl_gene.json.gz.Vertex.json: no such file or directory
Writing:  output/ensembl_transcript.json.gz
ERROR ON FILE CREATE
ERROR ON FILE CREATE
ERROR ON FILE CREATE
LENGTH OF PROCCHAN 3graph schema validate error at
graph schema validate error at
graph schema validate error at
graph schema validate error at
graph schema validate error at
graph schema validate error at
Error getting file information: stat output/ensembl_transcript.json.gz.OutEdge.json: no such file or directory
Error getting file information: stat output/ensembl_transcript.json.gz.InEdge.json: no such file or directory
Error getting file information: stat output/ensembl_transcript.json.gz.Vertex.json: no such file or directory```

**Comment by kellrott** [comment]
The go.mod / go.sum files are way to big. Try running `go mod tidy`

**Comment by kellrott** [comment]
The included command line program should parse command arguments, rather then having paths hard coded.
I would recommend looking at https://github.com/spf13/cobra

File: cmd/gengraph/main.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+	for scanner.Scan() {
+		count++
+	}
+
+	return count
+}
+
+// https://github.com/bmeg/sifter/blob/51a67b0de852e429d30b9371d9975dbefe3a8df9/transform/graph_build.go#L86
+var Cmd = &cobra.Command{
+	Use:   "gen-graph",
```
Comment (kellrott) [inline cmd/gengraph/main.go]: Add some command line annotations, so the help is more descriptive, like:
```
Use:   "gen-graph [schema dir] [input file] [class] [outdir]",
```

File: util/loader.go [hunk 2]
```diff
@@ -52,89 +52,154 @@ func yamlLoader(s string) (io.ReadCloser, error) {
...
+	templatePointer map[string]any
 }
 
 type GraphExtension struct {
 	Targets []Target
 }
 
 func (s GraphExtension) Validate(ctx jsonschema.ValidationContext, v interface{}) error {
-	//fmt.Printf("graph schema validate\n")
+	fmt.Println("graph schema validate error at ", v)
```
Comment (kellrott) [inline util/loader.go]: This seems to produce a lot of extra non-error messages that look like errors

Comment (matthewpeterkort) [reply util/loader.go]: commented that out in newer commits

File: cmd/gengraph/main.go (hunk 1 above)
Comment (matthewpeterkort) [reply cmd/gengraph/main.go]: updated

**Pull request merged**